apps/aecore/lib/aecore/chain/worker.ex

```diff
@@ -506,7 +506,8 @@ defmodule Aecore.Chain.Worker do
 
     if Enum.empty?(blocks_map) do
       [block_hash] = Map.keys(state.blocks_data_map)
-      Persistence.add_block_by_hash(block_hash, state.blocks_data_map[block_hash])
+      %{block: block} = state.blocks_data_map[block_hash]
+      Persistence.add_block_by_hash(block_hash, block)
```

Comment:
why does get... lead to add... ?

Reply:
this does not return header right now?

Reply:
1st Q: I'm not quite sure why, it was like this in master already;
2nd Q: The problem was that we added in the persistence the whole `%{block: block}` map instead of just adding the block itself;

Reply:
1. okay lets keep it for now
2. but you can return `.header` right?

Reply:
Yes, it's inside the `block` map data

Reply:
So if it supposed to return the header, do so